jenkins/infra/terraform/templates/user_data.tpl

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -xu
+
+BACKUP_DIR="${backup_dir}"
+BACKUP_BUCKET="${backup_bucket}"
+JENKINS_BACKUP_DMS="${jenkins_backup_dms}"
+
+die() {
+    echo "$*" 1>&2
+    exit 1
+}
+
+restore-backup-set() {
+    systemctl stop jenkins
+    echo "Restoring from backup"
+    aws s3 sync --delete --exclude=.initial-sync "s3://$BACKUP_BUCKET/" "$BACKUP_DIR/" --quiet
+    chown -R jenkins:jenkins "$BACKUP_DIR"
+    touch "$BACKUP_DIR/.initial-sync"
+    systemctl start jenkins
+}
+
+ci-restore() {
+    systemctl stop jenkins
+    # List all backups in proper order
+    ALL_BACKUPS=$(find $$BACKUP_DIR -maxdepth 1 -type d   -name 'FULL*' -o -name 'DIFF*' | sort -t- -k2)
+
+    # Find the last full backup
+    LAST_FULL=$(basename "$(echo "$$ALL_BACKUPS" | grep FULL | tail -n1)")
+
+    # And all following incrementals
+    INCREMENTALS=$(echo "$$ALL_BACKUPS" | sed -e "0,/$LAST_FULL/d" | xargs -n1 basename)
+
+    # Recover from latest backup (full + incrementals)
+    for BACKUP in $$LAST_FULL $$INCREMENTALS; do
+        echo "Restoring from $$BACKUP_DIR/$BACKUP/"
+        su - jenkins -c "rsync -av $BACKUP_DIR/$BACKUP/ /var/lib/jenkins/"
+    done
+    systemctl start jenkins
+}
+
+lock() {
+    local prefix=$$1
+    local lock_file="/var/lock/$$prefix.lock"
+
+    # create lockfile
+    eval "exec 200>$$lock_file"
```

Comment:
is the `$$` intended on this line, seems like the pid was already captured on line 44.

Reply:
The `$$` in this case does not mean pid in bash, I'm trying to escape the `$`

From terraform's documentation

> You can escape interpolation with double dollar signs: $${foo} will be rendered as a literal ${foo}.

https://www.terraform.io/docs/configuration/interpolation.html